services/web/README.md

```diff
@@ -24,3 +24,114 @@ When defined via the `WEB_UI_CONFIG_FILE` environment variable, the configuratio
 ### Embedding Web
 
 Web can be consumed by another application in a stripped down version called “Embed mode”. This mode is supposed to be used in the context of selecting or sharing resources. For more details see the developer documentation [ownCloud Web / Embed Mode](https://owncloud.dev/clients/web/embed-mode/). See the environment variables: `WEB_OPTION_MODE` and `WEB_OPTION_EMBED_TARGET` to configure the embedded mode.
+
+# Web Apps
+
+The administrator of the environment is capable of providing custom web applications to the users.
+This feature is useful for organizations that have specific web applications that they want to provide to their users.
+
+It's important to note that the feature at the moment is only capable of providing static (js, mjs, e.g.) web applications
+and does not support injection of dynamic web applications (custom dynamic backends).
+
+## Loading Applications
+
+Loading applications is provided in three ways:
+
+* By loading a web application from a user-provided path, by setting the `WEB_APPS_PATH` environment variable.
+* By loading a web application from the default ocis home directory, e.g. `$OCIS_BASE_DATA_PATH/web/assets/apps`.
+* By embedding a web application into the ocis binary which is a build-time option,
+  e.g. `ocis_src_path/services/web/assets/apps` followed by a build.
+
+The list of available applications is composed of the build in extensions and the custom applications
+provided by the administrator, e.g. `WEB_APPS_PATH` or `$OCIS_BASE_DATA_PATH/web/assets/apps`.
+
+for example, if ocis contains a build in extension `image-viewer-dfx` and the administrator provides a custom
+application `image-viewer-obj` in the `WEB_APPS_PATH` directory,the user will be able to access both applications
+from the web ui.
+
+## Application Structure
+
+Applications always have to follow a strict structure, which is as follows:
+
+* an application must be a directory
+* an application must contain a `manifest.json` file
+
+everything else is skipped and not considered as an application.
+
+The `manifest.json` file contain the following fields:
+
+* `id` - required - the name of the application must be unique across all applications
```

Comment:
Could you please add where the id is coming from? Does it need to be equal to the folder name in the apps asset path? Or same id as in the appInfo object inside the entrypoint?

Edit: found the code path. Id needs to be equal to the folder name. Wouldn't it make more sense to read the folder name from the filesystem then and don't include the field in the manifest.json at all?